tests/checks/mock/test_fail2ban.py

```diff
@@ -0,0 +1,167 @@
+# stdlib
+import unittest
+import subprocess
+
+# third party
+import mock
+
+# project
+from tests.checks.common import AgentCheckTest, get_check
+
+
+class Fail2BanTestCase(unittest.TestCase):
```

Comment:
If you subclass `AgentCheckTest` instead you'll get a nicer interface for certain things. Details in subsequent comments